### PR TITLE
[Refactor] 2: Switch marketplace to coordinator

### LINKED ISSUE #243: [Builder State Refactor] Port marketplace builder to BSC

### PR DESCRIPTION
**Pull request by QuentinI**
Closes #243 

### This PR:
Ports over marketplace builder to the new builder state coordinator (see #237 )

### This PR does not:
Change the legacy builder

### Key places to review:
Everything.
`crates/marketplace/src/service.rs` is probably best read as a wholly new file rather than trying to read through the diff.

### CODE REVIEW

File: crates/marketplace/src/testing/integration.rs [hunk 1]
```diff
@@ -78,93 +67,34 @@ where
...
+            BUILDER_CHANNEL_CAPACITY,
+            1, // Arbitrary base fee
+            config.hooks,
         );
 
-        builder_state.event_loop();
-
-        let hooks = Arc::new(NoHooks(PhantomData));
-
         // create the proxy global state it will server the builder apis
```
Comment (shenkeyao) [inline crates/marketplace/src/testing/integration.rs:87]: Nit: After the test update, `app` is now built from a global state, so not a _proxy_ global state anymore.

File: crates/marketplace/src/testing/order_test.rs [hunk 2]
```diff
@@ -234,23 +224,20 @@ async fn test_builder_order() {
...
-                .unwrap();
+            for evt in events {
+                event_stream_sender
+                    .broadcast(Event {
+                        view_number: ViewNumber::new(round as u64),
+                        event: evt,
+                    })
+                    .await
+                    .unwrap();
+            }
```
Comment (shenkeyao) [inline crates/marketplace/src/testing/order_test.rs:240]: Nit: This seems to be the same in several places. Can we move the `broadcast` calls to the `calc_proposal_events` function (also rename the function)?

File: crates/marketplace/src/service.rs [hunk 3]
```diff
@@ -290,43 +66,64 @@ pub struct ProxyGlobalState<Types: NodeType, H: BuilderHooks<Types>> {
 
     // Maximum time allotted to wait for bundle before returning an error
     api_timeout: Duration,
+
+    maximize_txn_capture_timeout: Duration,
```
Comment (shenkeyao) [inline crates/marketplace/src/service.rs:70]: Nit: I feel like using the verb `maximize` is a bit strange, maybe `maximum` or just `max`.

File: crates/marketplace/src/service.rs [hunk 4]
```diff
@@ -337,22 +134,171 @@ where
...
+            hooks.handle_hotshot_event(&event).await;
+
+            match event.event {
+                EventType::Error { error } => {
+                    tracing::error!("Error event in HotShot: {:?}", error);
+                }
+                EventType::Transactions { transactions } => {
+                    let transactions = hooks.process_transactions(transactions).await;
+
+                    // TODO: record results
```
Comment (shenkeyao) [inline crates/marketplace/src/service.rs:172]: What do we want to record here?

**Review by shenkeyao — APPROVED** [review]
Looks really good. Just nit comments.

File: crates/marketplace/src/testing/integration.rs (hunk 1 above)
Comment (QuentinI) [reply crates/marketplace/src/testing/integration.rs:87]: Yeah, this is a poorly-worded comment. Changed.

File: crates/marketplace/src/testing/order_test.rs (hunk 2 above)
Comment (QuentinI) [reply crates/marketplace/src/testing/order_test.rs:240]: Sure! I've gone a step further and moved most of the proposal wrangling code to a separate type that keeps track of chain state.

File: crates/marketplace/src/service.rs (hunk 3 above)
Comment (QuentinI) [reply crates/marketplace/src/service.rs:70]: It's not really about the timeout here, it's about the process of "maximizing txn capture". Nonetheless, it doesn't look like a good variable name, I've renamed it and added a doc comment for clarity

File: crates/marketplace/src/service.rs (hunk 4 above)
Comment (QuentinI) [reply crates/marketplace/src/service.rs:172]: I remember I was thinking about the transaction status API, no idea why I'd add the comment though, as it has no relation to this PR 😅 . Removed it.

File: crates/marketplace/src/testing/order_test.rs (hunk 2 above)
Comment (shenkeyao) [reply crates/marketplace/src/testing/order_test.rs]: Good change!

**Review by shenkeyao — APPROVED** [review]

**Pull request merged**